### PR TITLE
DYN-1078 Prevent `-` Button Removing Required Ports

### PR DESCRIPTION
**Pull request by StudioLE**
### Purpose

JIRA: [DYN-1078]( https://jira.autodesk.com/browse/DYN-1078)

Issue: https://github.com/DynamoDS/Dynamo/issues/8741

Prevent the user from being able to remove default ports using the `-` button on `DSVarArgFunction` nodes

Behaviour before:

DYN-1078-Remove-Default-Inputs-Before-v2

Behaviour after:

DYN-1078-Remove-Default-Inputs-Completed

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@mjkkirschner @ QilongTang @mmisol

### FYIs

@Amoursol

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@StudioLE  - thanks for the fix- can you also try adding a test that calls the `RemoveInputFromModel` method and asserts the number of ports is still equal to the default?

**Comment by mjkkirschner** [comment]
@StudioLE - unfortunately there are 4 failing tests now:
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/1861/

@SHKnudsen should have access to this dashboard if you do not -

can you try these out locally?

**Comment by SHKnudsen** [comment]
Thanks @mjkkirschner, we'll have a look.

**Comment by StudioLE** [comment]
Thanks @mjkkirschner. This is now resolved with 70f0e24.

image

**Review by mjkkirschner — APPROVED** [review]

File: src/DynamoCore/Graph/Nodes/VariableInputNode.cs [hunk 1]
```diff
@@ -173,14 +174,16 @@ private void MarkNodeDirty()
...
+            // or if the node is a DSVarArgFunction and doing so would remove a default port
+            bool countIsGreaterThanDefault = !(model is DSVarArgFunction dSVarArgFunction) || count > dSVarArgFunction.DefaultNumInputs;
+            if (count != 0 && countIsGreaterThanDefault)
             {
                 var port = model.InPorts[count - 1];
                 port.DestroyConnectors();
                 model.InPorts.Remove(port);
             }
-
             MarkNodeDirty();
```
Comment (QilongTang) [inline src/DynamoCore/Graph/Nodes/VariableInputNode.cs:187]: @StudioLE @mjkkirschner One thing I found from my local testing of this is that, even with the default state and user would not delete default ports, but node still marked dirty which I think is not necessary because that will trigger a graph run. Can we make another PR and put the statement in the `if` above?
